### ISSUE TITLE
能在windows环境下实现吗

### ISSUE THREAD

**Issue by chaoqunWang666** [DonaldRR/SimpleNet#74]

**Comment by dhyuk54** [comment]
能  下载数据集之后 参考sh文件放到指定的目录就行,如果 你想调试的话 把sh文件复制之后问ChatGPT 然后加在输入参数里了 就能调试了

**Comment by Occupied-By-Aurora** [comment]
![image](https://github.com/user-attachments/assets/98d94531-65fb-4301-a98f-ab6c2171f1ba)
Why do I put the parameters in the run file into the main file parameters, and then run the main file, the program ends directly, I do not see the process of training!

**Comment by Occupied-By-Aurora** [comment]
![image](https://github.com/user-attachments/assets/e2cae31a-f540-42bc-9bcb-cc393c2e0bf0)
I see no action in main, what should I write?

**Comment by dhyuk54** [comment]
pycharm -> edit -> shell script->
--gpu
0
--seed
0
--log_group
simplenet_mvtec
--log_project
MVTecAD_Results
--results_path
results
--run_name
run
net
-b
wideresnet50
-le
layer2
-le
layer3
--pretrain_embed_dimension
1536
--target_embed_dimension
1536
--patchsize
3
--meta_epochs
4
--embedding_size
256
--gan_epochs
4
--noise_std
0.015

然后加上自己的MVTec的数据集

**Comment by chaoqunWang666** [comment]
谢谢大家！！！

**Comment by chaoqunWang666** [comment]
已经跑成功了！！

**Comment by Occupied-By-Aurora** [comment]
你是设置了什么吗，还是就是按照上面说的配置重新配置了.sh脚本文件？


> 已经跑成功了！！

**Comment by Occupied-By-Aurora** [comment]
> pycharm -> edit -> shell script-> --gpu 0 --seed 0 --log_group simplenet_mvtec --log_project MVTecAD_Results --results_path results --run_name run net -b wideresnet50 -le layer2 -le layer3 --pretrain_embed_dimension 1536 --target_embed_dimension 1536 --patchsize 3 --meta_epochs 4 --embedding_size 256 --gan_epochs 4 --noise_std 0.015
> 
> 然后加上自己的MVTec的数据集

![image](https://github.com/user-attachments/assets/0a22e9f4-e0a8-4dca-93fc-43a4ebb4e481)
我运行了还是不行，想问一下还有哪里不对吗

**Comment by Occupied-By-Aurora** [comment]
我不太习惯用命令行运行，想问一下如果调试的话，又应该怎么设置呢main文件呢，主函数里面要加程序吗

**Comment by dhyuk54** [comment]
> > pycharm -> edit -> shell script-> --gpu 0 --seed 0 --log_group simplenet_mvtec --log_project MVTecAD_Results --results_path results --run_name run net -b wideresnet50 -le layer2 -le layer3 --pretrain_embed_dimension 1536 --target_embed_dimension 1536 --patchsize 3 --meta_epochs 4 --embedding_size 256 --gan_epochs 4 --noise_std 0.015
> > 然后加上自己的MVTec的数据集
> 
> ![image](https://private-user-images.githubusercontent.com/124340272/393489518-0a22e9f4-e0a8-4dca-93fc-43a4ebb4e481.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NTg1OTIsIm5iZiI6MTczMzU1ODI5MiwicGF0aCI6Ii8xMjQzNDAyNzIvMzkzNDg5NTE4LTBhMjJlOWY0LWUwYTgtNGRjYS05M2ZjLTQzYTRlYmI0ZTQ4MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwN1QwNzU4MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mOWQxNGNmNTFhNTljZmY5NjliOTg3ZmJkNDUxMDRjMmQ3MDYwOWZkMTg0NDI3Y2ZkMTk3ZGUwODU4YWI1NGQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Ewfk4K3uD1vkR7ySmMbiJzxNnNuSAjBuJOY0GKunlFU) 我运行了还是不行，想问一下还有哪里不对吗

哎呀我的天那 你这基本功不扎实啊 sh 文件你运行了之后 是不能debug的 所以 把.sh文件的命令 输入了 pycharm的 debug script当中  跑开源代码 无非就是 你的 python  torch版本装的对不对 还有就是数据集路径的问题了 实在不行 你就租一个linux机子 跑一下  然后 改成 windows在跑跑试试

**Comment by dhyuk54** [comment]
> 我不太习惯用命令行运行，想问一下如果调试的话，又应该怎么设置呢main文件呢，主函数里面要加程序吗

这么写有利于维护  只修改配置文件参数即可 工程上都习惯用.sh文件作为配置参数   你看看 import click 模块是怎么玩的 你就知道入口函数是什么了

**Comment by Occupied-By-Aurora** [comment]
> > 我不太习惯用命令行运行，想问一下如果调试的话，又应该怎么设置呢main文件呢，主函数里面要加程序吗
> 
> 这么写有利于维护 只修改配置文件参数即可 工程上都习惯用.sh文件作为配置参数 你看看 import click 模块是怎么玩的 你就知道入口函数是什么了

好的，十分感谢

**Comment by ample52004** [comment]
请教一下具体要怎么输入，python3 main.py --gpu 0 --seed 0 --log_group simplenet_mvtec --log_project MVTecAD_Results --results_path results --run_name run net -b wideresnet50 -le layer2 -le layer3 --pretrain_embed_dimension 1536 --target_embed_dimension 1536 --patchsize 3 --meta_epochs 4 --embedding_size 256 --gan_epochs 4 --noise_std 0.015  -d screw -d pill -d capsule -d carpet -d grid -d tile -d wood -d zipper -d cable -d toothbrush -d transistor -d metal_nut -d bottle -d hazelnut -d leather mvtec /data/MVTec_ad
我尝试输入以下命令模型不运行
